mailbox/store/src/test/java/org/apache/james/mailbox/store/MessageBuilder.java

```diff
@@ -45,7 +59,26 @@ public MailboxMessage build() throws Exception {
     }
 
     public MailboxMessage build(MessageId messageId) throws Exception {
-        return new SimpleMailboxMembership(messageId, mailboxId, uid, -1,  internalDate, size, flags, body, headers);
+        byte[] headerContent = getHeaderContent();
+        SimpleMailboxMessage mailboxMessage = new SimpleMailboxMessage(messageId, internalDate, size, headerContent.length,
+            new SharedByteArrayInputStream(Bytes.concat(headerContent, body)), flags, new PropertyBuilder(), mailboxId, NO_ATTACHMENTS);
+        mailboxMessage.setUid(uid);
+        return mailboxMessage;
+    }
+
+    private byte[] getHeaderContent() throws IOException {
```

Comment:
This class is not tested AFAIK, the code you wrote (or probably copy/paste) is not proved to work, right?